src/newbank/server/NewBank.java

```diff
@@ -1,49 +1,128 @@
 package newbank.server;
 
-import java.util.HashMap;
+import java.util.Optional;
+import java.util.Set;
 
 public class NewBank {
-		
-	private customerDB customers;
-	
-	public NewBank() {
-		customers = new customerDB();
-	}
-	
-	public synchronized Customer checkLogInDetails(String customerID, String password) {
-		Customer customer = customers.getCustomer(customerID);
-		if (customer == null) {
-			return null;
-		} else if ( customer.checkPassword(password) ) {
-			return customer;
-		} else {
-			return null;
-		}
-	}
-
-	// commands from the NewBank customer are processed in this method
-	public synchronized String processRequest(String customerID, String request) {
-		Customer customer = customers.getCustomer(customerID);
-		if( customer != null ) {
-			if (request.equals("SHOWMYACCOUNTS")) {
-				return showMyAccounts(customer );
-			} else if (request.startsWith("NEWACCOUNT")) {
-				String accountName = request.substring( request.indexOf(" ")+1);
-				return makeNewAccount(customer, accountName);
-			}
-		}
-		return "FAIL";
-	}
-	
-	private String showMyAccounts(Customer customer) {
-		return customer.accountsToString();
-	}
-	
-	private String makeNewAccount(Customer customer, String accountName) {
-		customer.addAccount( new Account(accountName, 0.0) );
-		customers.updateCustomer(customer);
-		return "SUCCESS";
-	}
-	
 
+  private static final String SPACE = " ";
+
+  private CustomerDB customers;
+
+  public NewBank() {
+    customers = new CustomerDB();
+  }
+
+  public synchronized Customer checkLogInDetails(String customerID, String password) {
+    Customer customer = customers.getCustomer(customerID);
+    if (customer == null) {
+      return null;
+    } else if (customer.checkPassword(password)) {
+      return customer;
+    } else {
+      return null;
+    }
+  }
+
+  // commands from the NewBank customer are processed in this method
+  public synchronized String processRequest(String customerID, String request) {
+    Customer customer = customers.getCustomer(customerID);
+    if (customer != null) {
+      if (request.equals("SHOWMYACCOUNTS")) {
+        return showMyAccounts(customer);
+      } else if (request.startsWith("NEWACCOUNT")) {
+        String accountName = request.substring(request.indexOf(SPACE) + 1);
+        return makeNewAccount(customer, accountName);
+      } else if (request.startsWith("MOVE")) {
+        return moveMoney(customer, request.split(SPACE));
```

Comment:
Instead of single space regex for multiple space might be better. .split("\\s+") maybe. Our algorithm wont break with multiple spaces in that case. Just a suggestion

Reply:
Good point, I'm going to correct this.